import-automation/progress-dashboard-rest/app/service/base_database.py

```diff
@@ -0,0 +1,135 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Base class for a Database service that stores some kind of entities using
+Google Cloud Datastore for storage.
+"""
+
+from google.cloud import datastore
+
+from app import utils
+
+
+class BaseDatabase:
+    """Base class for a Database service that stores some kind of entities using
+    Google Cloud Datastore for storage.
+
+    New Entities must be created using get_by_id(make_new=True).
+
+    The datastore Client will attempt to infer credentials based on
+    host environment. See
+    https://cloud.google.com/docs/authentication/production#finding_credentials_automatically.
+
+    Attributes:
+        kind: Kind of entities to store
+        client: datastore Client to communicate with Datastore
+    """
+    def __init__(self, kind, client=None, id_field=None):
+        """Constructs an BaseDatabase.
+
+        Args:
+            kind: Kind of entities stored
+            client: Client to communicate with Datastore.
+            id_field: Name of the ID field of an entity as a string.
+                Every operation checks if id_field is not already a field in the
+                entity. If not, it will be added to the entity with the value of
+                the key name.
+        """
+        if not client:
+            client = utils.create_datastore_client()
+        self.kind = kind
+        self.client = client
+        self.id_field = id_field
+
+    def get_by_id(self, entity_id=None, make_new=False):
```

Comment:
The comparable dictionary lookup method is called "get".

Reply:
will fix